src/pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -186,6 +186,10 @@ def __init__(
         self._amp_level_flag: Optional[str] = amp_level
         self._auto_select_gpus: bool = auto_select_gpus
 
+        # handle "gpu"
+        if accelerator == "gpu":
+            accelerator = self._choose_gpu_accelerator_backend()
```

Comment:
Why do we introduce a special case handling here?
I believe this should go into the `self._choose_accelerator()` as this falls under the "auto" case. 
I would prefer to follow the high-level structure we have established in the connector to stay organized.

Reply:
The issue with that is that the accelerator-type is checked in `_check_config_and_set_final_flags` which is called before the actual auto-handling. This means that we still would require a special-casing there, which I tried to avoid by calling it earlier.

That was the main reason of https://github.com/Lightning-AI/lightning/pull/13642/commits/708b4b47c733a66fbd5d77abf4a7e7945e57d63f